src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.Resolution.cs

```diff
@@ -935,6 +935,11 @@ private static void AddModule(PEModule module, int referenceIndex, ResolvedRefer
             {
                 AssemblyIdentity definition = definitions[i].Identity;
 
+                if (reference.Name != definition.Name)
+                {
+                    continue;
+                }
+
                 switch (assemblyIdentityComparer.Compare(reference, definition))
```

Comment:
shoudl that check just move into `assemblyIdentityComparer.Compare` which woudl then return `.NotEquivalent` immediately?

Reply:
Very possibly, but right now I'm doing to the quickest, most local change I can to get a test insertion out :slightly_smiling_face:.